### PR TITLE
multiline string support

### PR DESCRIPTION
**Pull request by PoseidonCoder**
It now activates multiline mode when it encounters a multiline/triple-quoted string.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -120,7 +124,17 @@
...
+          if (formattedOut.indexOf('\n') == -1) {
+            /*
+            * output includes no newline characters
+            */
+            term.echo(`'${formattedOut}'`);
+          } else {
+            /*
+            * output includes newline characters
+            */
+            term.echo(`"""${formattedOut}"""`)
```
Comment (daniel-schroeder-dev) [inline index.html]: I can't seem to get this to fire. What input have you given it where this works?

File: index.html [hunk 2]
```diff
@@ -183,13 +197,8 @@
         *   If this is a single-line print statement, just execute it.
         */
         // line = `${}`
-      } else if (line.match(/^\w+$/m)) {
```
Comment (daniel-schroeder-dev) [inline index.html:186]: Why did you remove this? This is needed to get single-line statements to display correctly. In your version, this will display like a tuple:

```python
>>> a = [1, 2, 3]
>>> a
(1, 2, 3)
```

File: index.html [hunk 3]
```diff
@@ -242,10 +264,16 @@
               *   This just sticks the gutter in and delays processing until we exit
               *   the mutliline statement.
               */
-              if (multilineMode.mode === ML_FUNC) {
-                this.insert(multilineFuncGutter);
-              } else if (multilineMode.mode === ML_STRUCT) {
-                this.insert(multilineStructGutter);
+              switch (multilineMode.mode) {
```
Comment (daniel-schroeder-dev) [inline index.html:269]: I like this, much cleaner!

File: index.html [hunk 4]
```diff
@@ -202,35 +211,48 @@
...
-                *   This will run the callback to terminal() and then the entire 
-                *   statement will be passed in for processing.
-                */
-                return original();
-              } else {
-                this.insert(multilineFuncGutter);
-              }
-            } else if (multilineMode.mode === ML_STRUCT) {
-              if (line.match(/[}\]\)]$/)) {
+            switch (multilineMode.mode) {
```
Comment (daniel-schroeder-dev) [inline index.html]: I don't like switches for stuff with if...else logic nested inside. The `break` statement thing can easily get removed and cause nasty fall-through issues.

**Review by daniel-schroeder-dev — CHANGES_REQUESTED** [review]
I like the added functionality, but resolve the issues I pointed out before I can merge this in. Also, maybe add in single-quote multi-line support as well, since having only one would be odd? Good work though, and thanks for helping out with this!

File: index.html (hunk 1 above)
Comment (PoseidonCoder) [reply index.html]: It seems like this block was never executed, therefore it never worked. But, I moved it around to another block and now it works!

*I added that in the latest commit: https://github.com/daniel-schroeder-dev/cwhq-python-repl-hacking/pull/1/commits/76a68118b492b66dbc7a4d5e4867af19f1f86fe4*

File: index.html (hunk 4 above)
Comment (PoseidonCoder) [reply index.html]: Would you like me to convert it back to an if-else statement?

Comment (daniel-schroeder-dev) [reply index.html]: Yeah, I think it's safer as if...else.

File: index.html (hunk 2 above)
Comment (PoseidonCoder) [reply index.html:186]: I removed it because the RegEx would end up matching some of the line text.
*But*, in the recent commit, https://github.com/daniel-schroeder-dev/cwhq-python-repl-hacking/pull/1/commits/8e2898111491428863b120d9641a038ebd19e755, I added it back and fixed the issue.

File: index.html [hunk 5]
```diff
@@ -21,10 +21,12 @@
 
     const ML_FUNC = "ML_FUNC";
     const ML_STRUCT = "ML_STRUCT";
+    const ML_QUOTE = "ML_QUOTE";
 
     // var multilineModeEnabled = false;
     const multilineFuncGutter = "\n...\t\t";
     const multilineStructGutter = "\n... ";
+    const multilineQuoteGutter = "\n";
```
Comment (daniel-schroeder-dev) [inline index.html]: Use the same gutter as the `multilineStructGutter`. I want it to look like `bpython`:

```
>>> """
... sup
... dude
... """
```

File: index.html (hunk 4 above)
Comment (PoseidonCoder) [reply index.html]: Ok, I converted it https://github.com/daniel-schroeder-dev/cwhq-python-repl-hacking/pull/1/commits/f718a52e76105abc9d6238d0cf450818c6da71d0.

File: index.html (hunk 5 above)
Comment (PoseidonCoder) [reply index.html]: ok https://github.com/daniel-schroeder-dev/cwhq-python-repl-hacking/pull/1/commits/57db4946877aea6dda829c55488f176b67edabfd

File: index.html [hunk 6]
```diff
@@ -26,7 +26,7 @@
     // var multilineModeEnabled = false;
     const multilineFuncGutter = "\n...\t\t";
     const multilineStructGutter = "\n... ";
-    const multilineQuoteGutter = "\n";
+    const multilineQuoteGutter = "\n...";
```
Comment (daniel-schroeder-dev) [inline index.html]: Not quite! Add a space after the `...` and you're good though.

**Review by daniel-schroeder-dev — CHANGES_REQUESTED** [review]

**Comment by PoseidonCoder** [comment]
> I like the added functionality, but resolve the issues I pointed out before I can merge this in. Also, maybe add in single-quote multi-line support as well, since having only one would be odd? Good work though, and thanks for helping out with this!

Ok, I also added support for single-quote multi-line strings as well.

**Comment by daniel-schroeder-dev** [comment]
Great work! I have to fix something in the REGEX that removes the gutter, but I'll do that now. It should be doing this:

```
>>> s = """
... whoa
... """
>>> s

whoa

>>> 
```
But it's doing this:

```
>>> s = """
... whoa
... """
>>> s
 whoa
 
>>> 

```
